cmd/criticize/main.go

```diff
@@ -302,3 +311,41 @@ func shortenLocation(loc string) string {
 		return loc
 	}
 }
+
+func output(l *linter, rule, loc, warn string) {
+	switch {
+	case l.flags.JSONOutput:
+		// due to sort capabilities,
+		// using struct instead of map
+		b, err := json.MarshalIndent(struct {
+			Rule     string `json:"rule"`
+			Location string `json:"location"`
+			Warning  string `jdon:"warning"`
+		}{
+			Rule:     rule,
+			Location: loc,
+			Warning:  warn,
+		}, "", "    ") // TODO:
+		// Should add a prefix to add split capabilities to split the message to every single report
+		if err != nil {
+			panic(err)
+		}
+		log.Println(string(b))
+	case l.flags.ColoredOutput:
+		log.Printf("%v:\n%v: %v\n",
+			aurora.Red(rule),
+			aurora.Magenta(
+				aurora.Bold(loc),
+			),
+			warn,
+		)
+
+	default:
+		log.Printf("%s:\n%s: %s\n",
+			rule,
+			loc,
+			warn,
+		)
```

Comment:
Adding new formats might be okay, but why changing the default one?
There is no "right default format", and I personally don't like a newline inside a warning message.